### PR TITLE
Fix dev environment for new installs

### LINKED ISSUE #602: Fix dev environment for new installs
Vagrant installs are failing and so are some of the tests. These issues need to be addressed.

### PR DESCRIPTION
**Pull request by bbusenius**
Fixes #602 

**Changes in this request**
- Fixes `django-static-precompiler` package in `requirements.txt`
- Gets unit tests passing again
- Fixes `--parallel` testing in the Vagrant environment
- Allocates more memory and cpu to Virtualbox
- New Vagrantfile and dev version of the site that runs on Ubuntu jammy and `Python 3.9`.
- Tweaks to `requirements.txt` for `Python 3.9`

### CODE REVIEW

**Comment by bbusenius** [comment]
To test the build, you'll need to add this to your `local.py`:
```
WAGTAILSEARCH_BACKENDS = { 
    'default': { 
        'BACKEND': 'wagtail.search.backends.db',
    } 
}
```
I plan to get elasticsearch working in a followup issue.

**Comment by bbusenius** [comment]
I created #605 to deal with elasticsearch and any other improvements we might need.

File: Vagrantfile [hunk 1]
```diff
@@ -48,7 +48,22 @@ Vagrant.configure("2") do |config|
...
     # Customize the amount of memory on the VM:
-    vb.memory = "4096"
+    vb.memory = "10240"
+    vb.cpus = 8
+
+    up_message = <<-MSG
+
+    WRITE SOME CODE!!!
+         ___________________________            ____
+    ...  \____DLDC_220_________|_// __=*=__.--"----"--._________
```
Comment (bufordrat) [inline Vagrantfile:58]: This is so awesome.

**Review by bufordrat — APPROVED** [review]
Beautiful!  Can't get over that Star Trek ASCII art.  Builds a bit cleaner than it used to on my machine, I think.

**Review by bluairis — APPROVED** [review]
I also think that it builds cleaner and I like having the spaceship as confirmation

**Pull request merged**